src/GraphQL.Tests/Types/AutoGeneratedFieldRegistrationTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿using GraphQL.Types;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Types
+{
+    public class AutoGeneratedFieldRegistrationTests
+    {
+        private class TestPoco
+        {
+            public int UserId { get; set; } = 123;
+
+            public string UserName { get; set; } = "Sally";
+
+            public double UserRating { get; set; } = 1.0;
+        }
+
+        [Fact]
+        public void autogenerated_fields_should_exist()
+        {
+            var graphType = new AutoRegisteringObjectGraphType<TestPoco>();
+
+            graphType.HasField("UserId").ShouldBeTrue();
+            graphType.HasField("UserName").ShouldBeTrue();
+            graphType.HasField("UserRating").ShouldBeTrue();
+        }
+
+
+        [Fact]
+        public void should_not_throw_error_when_trying_to_register_field_with_same_name()
```

Comment:
Well I have 100's of tables and I like the idea of autogenerating the fields. But I like to have control over how certain fields resolve. Or remove them all together.

This is similar to what Hot Chocolate allows you to do. And was the one feature that I felt was missing from this project.

What part of this behaviout do you think would cause confusion. I feel like as long as its well documented it would be fine.

Reply:
sorry could you elaborate on what you mean by a method that handles this specific behaviour?

Do you mean i should do this instead

`
public class User
{
        public int UserId { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }      
 }  

// override the UserId name and resolver
GetField(f => f.UserId).Name("id").Resolve(c =>23456);

// remove the Lastname
RemoveField(f => f.Lastname);

//or you can do this
RemoveField("Lastname");
`

Reply:
FYI I did it the first way because i was trying to replicate how HotChocolate handles this scenario, but second way works as well